File: backend/pkg/models/asn.go
package models

import "time"

type ASN struct {
	ID             string    `json:"id" db:"id"`
	ASNNumber      int       `json:"asn_number" db:"asn_number"`
	Name           string    `json:"name,omitempty" db:"name"`
	OrganizationID string    `json:"organization_id,omitempty" db:"organization_id"`
	CountryCode    string    `json:"country_code,omitempty" db:"country_code"`
	CreatedAt      time.Time `json:"created_at" db:"created_at"`
}

File: android/core/model/src/main/java/io/ciphertun/ghi/core/model/Asn.kt
package io.ciphertun.ghi.core.model

import kotlinx.serialization.Serializable

@Serializable
data class Asn(
    val id: String,
    val asnNumber: Int,
    val name: String? = null,
    val organizationId: String? = null,
    val countryCode: String? = null,
)
